src/ContextEnricher/ContextEnricher.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Log\ContextEnricher;
+
+use InvalidArgumentException;
+use Yiisoft\Log\Message\CategoryFilter;
+
+/**
+ * @psalm-type Backtrace = list<array{
+ *      file:string,
+ *      line:int,
+ *      function?:string,
+ *      class?:string,
+ *      type?:string,
+ *  }>
+ */
+final class ContextEnricher implements ContextEnricherInterface
```

Comment:
Could it be named `ContextProvider`?

Reply:
Or `ContextProcessor`?

Reply:
This class don't provide context, but add info to context. For example we can use several classes via composite enricher.

Reply:
If move merge logic into `Logger`, then "provider" is good... I will try to do this.

Reply:
Done.